apps/web/components/MailingListSignup/MailingListSignup.tsx

```diff
@@ -0,0 +1,178 @@
+  const [status, setStatus] = useState<FormState>({
+    type: 'default',
+    message: '',
+  })
+
+  const handleSubmit = ({ email }: FormProps) => {
+    if (isValidEmail.test(email)) {
+      // Submit the valid email address.
+      const url = formatMailingSignupUrl(email)
+      jsonp(
+        url,
+        {
+          param: 'c',
+        },
+        (err: Error, data: any) => {
+          if (err) {
+            setStatus({
+              type: 'error',
+              message: err.message,
+            })
+          } else if (data.result !== 'success') {
+            setStatus({
+              type: 'error',
+              message: data.msg,
+            })
+          } else {
+            const successMessage: string = n(
+              'formSuccess',
+              'Skráning tókst. Þú þarft að fara í pósthólfið þitt og samþykkja umsóknina. Takk fyrir.',
+            )
+            setStatus({
+              type: 'success',
+              message: successMessage,
+            })
+          }
+        },
+      )
+    } else {
+      // Email address is not valid.
+      setStatus({
+        ...status,
+        type: 'error',
+        message: n('formInvalidEmail', 'Þetta er ógilt netfang.'),
+      })
+    }
+  }
+
+  /**
+   * Initialize Formik
+   */
+  const formikInitialValues: FormProps = { email: '' }
+  const formik = useFormik({
+    initialValues: formikInitialValues,
+    onSubmit: handleSubmit,
+  })
+
+  useEffect(() => {
+    if (status.type === 'error' && formik.values.email === '') {
+      // Reset the form to default state when the User clears the
+      // email input field after getting an error.
+      setStatus({
+        ...status,
+        type: 'default',
+        message: '',
+      })
+    }
+  }, [status.type, formik.values.email])
+
+  /**
+   * Formats the Mailing list signup URL with the given email.
+   *
+   * @param email A valid email address.
+   * @returns A URL to call to subsribe the given email address to the Mailing List.
+   */
+  const formatMailingSignupUrl = (email: string) => {
+    const encodedEmail = encodeURIComponent(email)
+    return mailingListUrl.replace(
+      mailingListUrlEmailReplacementToken,
+      encodedEmail,
+    )
+  }
+
+  /**
+   * Parses error messages to make them more user friendly.
+   *
+   * @param message An error message.
+   * @returns A more user friendly version of the given error message.
+   */
+  const parseErrorMessage = (message: string) => {
+    if (!message) {
+      return
+    }
+    const msg = message.toLowerCase()
+
+    if (
+      msg.includes('is already subscribed') ||
+      msg.includes('er nú þegar skráður')
+    ) {
+      return n('formEmailAlreadyRegistered', 'Þetta netfang er þegar á skrá.')
+    }
+
+    if (msg.includes('invalid email') || msg.includes('ógilt netfang')) {
+      return n('formInvalidEmail', 'Þetta er ógilt netfang.')
+    }
+
+    // TODO: Is this the proper way to log unexpected cases of error messages?
```

Comment:
@ingolfurs @Ingimarsson 
What is the proper way of logging warnings, such as this one, where unexpected/unhandled cases of the message occur? (Currently, I'm simply logging a warning to console.)